lib/complex.ex

```diff
@@ -433,11 +531,20 @@ defmodule Complex do
   [ln/1](#ln/1), [log10/1](#log10/1)
 
   #### Examples
-      iex> Complex.pow( Complex.fromPolar(2,:math.pi), Complex.imag() )
+      iex> Complex.power( Complex.from_polar(2,:math.pi), Complex.imag() )
       %Complex{im: 0.027612020368333014, re: 0.03324182700885666}
   """
-  @spec pow(complex, complex) :: complex
-  def pow(x = %Complex{}, y = %Complex{}) do
+  @spec power(complex, complex) :: complex
+  @spec power(number, complex) :: complex
+  @spec power(complex, number) :: complex
+  @spec power(number, number) :: number
+
+  def power(x, y) when is_number(x) and is_number(y), do: :math.pow(x, y)
```

Comment:
This will convert `power(integer, integer)` to float. You want to check if the first argument is an integer and the second is non negative and, if so, use Integer.pow/2.